base/src/Data/Macaw/Discovery.hs

```diff
@@ -898,6 +900,22 @@ stripPLTRead readId next rest =
         ArchState{} -> cont
         Comment{} -> cont
         _ -> Nothing
+  where
+    -- It is possible for later ArchState updates to reference the AssignId of
+    -- the AssignStmt that is dropped, so make sure to prune such updates to
+    -- avoid referencing the now out-of-scope AssignId.
+    dropRefsTo :: AssignId ids tp -> Stmt arch ids -> Stmt arch ids
+    dropRefsTo stmtId stmt =
+      case stmt of
+        ArchState addr updates ->
+          ArchState addr $
+          MapF.filter (\v -> Some stmtId `Set.notMember` refsInValue v) updates
+        AssignStmt{}       -> stmt
```

Comment:
I'm not sure if we need to worry about stale references appearing in these other forms of `Stmt`. (More generally, I'm not sure of the best way to test these changes.)

Reply:
`Comment` and `InstructionStart` are fine because they don't carry any values with them.

`stripPLTRead` bails out if it finds any of the others, so it should be fine (i.e., you shouldn't see any relevant things in `dropRefsTo`)

Reply:
Ah, right. I'll add some comments to this effect.

As far as testing goes, the test suite for `macaw-x86` seems to the best fit, as it has a collection of binaries that we could add to. But we'd likely need to fix #234 before we could successfully analyze such a binary.